File: app/src/main/java/com/drdisagree/iconify/receiver/BootReceiver.java
package com.drdisagree.iconify.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.drdisagree.iconify.services.UpdateScheduler;

public class BootReceiver extends BroadcastReceiver {

    private final String TAG = getClass().getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Log.i(TAG, "Broadcast received: " + intent.getAction());
            UpdateScheduler.scheduleUpdates(context);
        }
    }
}

File: app/src/main/java/com/drdisagree/iconify/receiver/BootReceiver.kt
package com.drdisagree.iconify.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.drdisagree.iconify.common.Const.ACTION_BOOT_COMPLETED
import com.drdisagree.iconify.common.Const.SYSTEMUI_PACKAGE
import com.drdisagree.iconify.services.UpdateScheduler

class BootReceiver : BroadcastReceiver() {

    private val tag = this::class.java.simpleName

    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            Log.i(tag, "Broadcast received: " + intent.action)

            // Schedule updates
            UpdateScheduler.scheduleUpdates(context)

            // Update QS clock on boot
            val broadcast: Intent = Intent(ACTION_BOOT_COMPLETED)
            broadcast.putExtra("packageName", SYSTEMUI_PACKAGE)
            context.sendBroadcast(broadcast)
        }
    }
}